pyproject.toml

```
@@ -1,57 +1,85 @@
+[build-system]
+build-backend = "uv_build"
+
+requires = [ "uv-build>=0.8.0,<0.9.0" ]
+
 [project]
 name = "postmodern-ci"
 version = "0.1.0"
-description = "Add your description here"
+description = "Example of a GitHub Actions-based CI and release pipelines with Python code via `uv`."
 readme = "README.md"
 authors = [
-    { name = "Nathan McDougall", email = "nathan.j.mcdougall@gmail.com" },
+  { name = "Nathan McDougall", email = "nathan.j.mcdougall@gmail.com" },
 ]
 requires-python = ">=3.12"
-dependencies = []
+classifiers = [
+  "Development Status :: 1 - Planning",
+  "Programming Language :: Python :: 3 :: Only",
+  "Programming Language :: Python :: 3.12",
+  "Programming Language :: Python :: 3.13",
+]
 
-[build-system]
-requires = ["hatchling"]
-build-backend = "hatchling.build"
+dependencies = [  ]
 
-[tool.uv]
-dev-dependencies = [
-    "coverage[toml]>=7.6.4",
-    "pre-commit>=4.0.1",
-    "pytest-cov>=5.0.0",
-    "pytest-md>=0.2.0",
-    "pytest>=8.3.3",
-    "ruff>=0.7.1",
-    "usethis",
+[dependency-groups]
+dev = [
+  "deptry>=0.23.0",
+  "pre-commit>=4.2.0",
+  "ruff>=0.7.1",
+]
+test = [
+  "coverage[toml]>=7.9.2",
+  "pytest>=8.4.1",
+  "pytest-cov>=6.2.1",
+  "pytest-md>=0.2.0",
 ]
 
-[tool.uv.sources]
-usethis = { path = "../usethis-python" }
+[tool.ruff]
+line-length = 88
+src = [ "src" ]
+format.docstring-code-format = true
 
-[tool.pytest.ini_options]
-testpaths = ["tests"]
-addopts = ["--import-mode=importlib"]
+lint.select = [
+  "C4",
+  "E4",
+  "E7",
+  "E9",
+  "F",
+  "FURB",
+  "I",
+  "PLE",
+  "PLR",
+  "PT",
+  "RUF",
+  "SIM",
+  "UP",
+]
 
-[tool.coverage.run]
-source = ["src"]
-omit = ["*/pytest-of-*/*"]
+[tool.codespell]
+ignore-regex = [ "[A-Za-z0-9+/]{100,}" ]
+ignore-words-list = [ "..." ]
 
-[tool.ruff]
-src = ["src"]
-line-length = 88
+[tool.pyproject-fmt]
+keep_full_version = true
+
+[tool.pytest.ini_options]
+testpaths = [ "tests" ]
+addopts = [ "--import-mode=importlib" ]
```

I think we use something like this

```suggestion
testpaths = [ "tests", "src" ]
addopts = [ "--import-mode=importlib", "--doctest-modules" ]
```